### PR TITLE
refactor: Migrate `admin/permissions` endpoints to tapir (DEV-1590)

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] build/chore: maintenance tasks (no production code change)
- [ ] docs: documentation changes (no production code change)
- [ ] feat: represents new features
- [ ] fix: represents bug fixes
- [ ] perf: performance improvements
- [x] refactor: represents production code refactoring
- [ ] test: adding or refactoring tests (no production code change)

### Basic requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

File: integration/src/test/scala/org/knora/webapi/util/ZioScalaTestUtil.scala [hunk 1]
```diff
@@ -18,4 +18,11 @@ object ZioScalaTestUtil {
     case Exit.Failure(err) => err.squash shouldBe a[T]
     case _                 => Assertions.fail(s"Expected Exit.Failure with specific T.")
   }
+  def assertFailsWithA[T <: Throwable: ClassTag](actual: Exit[Throwable, _], expectedError: String) = actual match {
+    case Exit.Failure(err) => {
+      err.squash shouldBe a[T]
+      err.squash.getMessage shouldEqual (expectedError)
+    }
+    case _ => Assertions.fail(s"Expected Exit.Failure with specific T.")
```
Comment (BalduinLandolt) [inline integration/src/test/scala/org/knora/webapi/util/ZioScalaTestUtil.scala:26]:
```suggestion
    case _ => Assertions.fail(s"Expected Exit. Failure with specific T.")
```

**Review by BalduinLandolt — APPROVED** [review]
Not the most detailed review from my side, but overall looks good, much cleaner than before. (Spotted one typo)

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/PermissionsEndpoints.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    .in(permissionsBase / "ap" / projectIri)
+    .description("Get all administrative permissions for a project.")
+    .out(sprayJsonBody[AdministrativePermissionsForProjectGetResponseADM])
+
+  val getPermissionsApByProjectAndGroupIri = base.securedEndpoint.get
+    .in(permissionsBase / "ap" / projectIri / groupIri)
+    .description("Get all administrative permissions for a project and a group.")
+    .out(sprayJsonBody[AdministrativePermissionGetResponseADM])
+
+  val getPermissionsDaopByProjectIri = base.securedEndpoint.get
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/api/PermissionsEndpoints.scala:53]: What `daop` means? This is probably subject to adjust for later, but I think it's wrongly abbreviated.

File: webapi/src/main/scala/org/knora/webapi/slice/admin/domain/model/GroupIri.scala [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import org.knora.webapi.slice.admin.domain.model.KnoraProject.Shortcode
+
+final case class GroupIri private (value: String) extends AnyVal
+
+object GroupIri {
+
+  implicit val tapirCodec: Codec[String, GroupIri, CodecFormat.TextPlain] =
+    Codec.string.mapEither(GroupIri.from)(_.value)
+
+  private val builtInGroups = List(
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/slice/admin/domain/model/GroupIri.scala:24]: This list exists in `OntologyConstans` line 496.

**Review by mpro7 — APPROVED** [review]
Only 2 minor comments you can ignore for now.

File: webapi/src/main/scala/org/knora/webapi/slice/admin/api/PermissionsEndpoints.scala (hunk 2 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/slice/admin/api/PermissionsEndpoints.scala]: You are right: Should be `doap` for `DefaultObjectAccessPermission` the same as the path segment this endpoint contains.

Fixed in 0f54648eff136381f4ec039a4f82f134c41bb59d

**Pull request merged**